### PR TITLE
Deps: un-pin (and avoid) rufus-scheduler

### PR DESCRIPTION
**Pull request by kares**
This reverts commit e6520cfee197a9fdcaa5e11e5081be06f81ca635 (https://github.com/elastic/logstash/pull/14150) and avoids using rufus-scheduler in x-pack.

## Release notes

[rn:skip]

## What does this PR do?

Removes the `'rufus-scheduler', '~> 3.0.9'` requirement from LS core.
This is now controlled by the scheduler [mixin](https://github.com/logstash-plugins/logstash-mixin-scheduler)
And LS should be good to go upgrading rufus to latest (3.8.1 atm).

## Why is it important/What is the impact to the user?

Newer rufus is expected to improve user experience e.g. no more `Fixnum` warnings printed.

## How to test this PR locally

check *Gemfile.lock* includes updated *rufus-scheduler* gem (>= 3.8.1)

## Related issues

- https://github.com/elastic/logstash/issues/13773

### CODE REVIEW

**Comment by kares** [comment]
@yaauie turns out we have another use of `Rufus::Scheduler` within core - which isn't a plugin.
instead of adapting the plugin-mixin to be include-able into a non-plugin, which sounds weird, I went ahead and simply avoided rufus in favor of a relatively simple (and similar) `TimerTask`, any concerns? :pray:

**Comment by kares** [comment]
> I went ahead and simply avoided rufus in favor of a relatively simple (and similar) TimerTask, any concerns?

just answering myself - there's already a precedent with using the `TimerTask` e.g. in [s3 output](https://github.com/logstash-plugins/logstash-output-s3/blob/v4.3.7/lib/logstash/outputs/s3.rb#L328-L332=)

> Do you think we can add a comment in logstash-core.gemspec with the link of meta issue to remind that avoid using rufus-scheduler?

thanks, went ahead and did not - it's a bit long explaining all the details in a code comment ...
let me know if you have concerns or you rather feel like we should remove the comment :pray:

File: logstash-core/logstash-core.gemspec [hunk 1]
```diff
@@ -77,11 +77,13 @@ Gem::Specification.new do |gem|
...
-  # Pin pending migration to shared Scheduler Mixin that can mitigate this issue.
-  # https://github.com/logstash-plugins/logstash-mixin-scheduler/pull/1
-  gem.add_runtime_dependency 'rufus-scheduler', '~> 3.0.9' #(MIT license)
+
+  # NOTE: plugins now avoid using **rufus-scheduler** directly, if logstash-core would find itself in a need
+  # to use rufus than preferably the **logstash-mixin-scheduler** should be changed to work with non-plugins.
+  #
+  # Using the scheduler directly might lead to issues e.g. when join-ing, see:
+  # https://github.com/logstash-plugins/logstash-mixin-scheduler/blob/v1.0.1/lib/logstash/plugin_mixins/scheduler/rufus_impl.rb#L85=
+  # and https://github.com/elastic/logstash/issues/13773
```
Comment (kaisecheng) [inline logstash-core/logstash-core.gemspec:86]: nice comment!

**Review by kaisecheng — APPROVED** [review]
LGTM

**Pull request merged**